fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -3117,7 +3159,10 @@ private boolean areAllRecordTypesSince(@Nullable Collection<RecordType> recordTy
             }
         }
 
-        return markIndexReadable(index).thenApply(b -> null);
+        // The index states is supposed to be empty (i.e. readable) because this is a new index. But unfortunately, the
+        // index state is keyed by index name which can be reused. So we need to mark the index readable. In such case,
+        // the built range is unlikely to be full so we should force it.
+        return markIndexReadable(index, true).thenApply(b -> null);
```

Comment:
There's something about having this catch and then ignore the error that feels a little strange....

Given that we're going to ignore the results of doing the validation, it almost seems like we should call `uncheckedMarkIndexReadable`. I could see alternative arguments that we could, like, check there are no uniqueness violations or data in the index subspace or something and then mark it as `READABLE` (perhaps only in the error case) (and maybe it also should add the `\x00` to `\xff` range to the build space?).

I could also see an argument that if it encounters an error, it should mark the index as `DISABLED` and wait for a rebuild. Then we don't get the index right away (sad) but we also no longer block `checkVersion`.

Something about the extra flag to force marking it as readable doesn't quite seem right.

Reply:
I didn't use `uncheckedMarkIndexReadable` mainly because I'd like it check and log exceptions even if not throw an exception.

I think " mark the index as DISABLED and wait for a **(manual, force)** rebuild" is an option. I just need to remove all my changes in `markIndexReadable` it will work like that. But that depends on how we see such error. If we think consider it as expected (even though not pleasant) as a side effect of reusing index name, I think it should better be working without manual intervention.

Reply:
Force build is not necessary. Just manual. So I think I'm fine with not marking index as readable, especially considering there may be underlying bugs that we need to be aware of.